spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestStoragePartitionedJoins.java

```diff
@@ -133,7 +133,44 @@ public void removeTables() {
     sql("DROP TABLE IF EXISTS %s", tableName(OTHER_TABLE_NAME));
   }
 
-  // TODO: add tests for truncate transforms once SPARK-40295 is released
+  // TODO: Truncate is not supported by SPJ yet (even after SPARK-40295).
+  // Add tests for full support once SPARK-50593 is resolved.
+  @TestTemplate
+  public void testJoinsWithIncompatibleTruncateSpecs() {
+
+    sql(
+        "CREATE TABLE %s (id BIGINT, int_col INT, dep STRING) "
+            + "USING iceberg "
+            + "PARTITIONED BY (truncate(4, dep)) "
+            + "TBLPROPERTIES (%s)",
+        tableName, tablePropsAsString(TABLE_PROPERTIES));
+
+    sql("INSERT INTO %s VALUES (1L, 100, 'software')", tableName);
+    sql("INSERT INTO %s VALUES (2L, 200, 'software')", tableName);
+    sql("INSERT INTO %s VALUES (3L, 300, 'software')", tableName);
+
+    sql(
+        "CREATE TABLE %s (id BIGINT, int_col INT, dep STRING) "
+            + "USING iceberg "
+            + "PARTITIONED BY (truncate(4, dep)) "
```

Comment:
The test name says `...IncompatibleTruncateSpecs` but both tables use the same partitioning `(truncate(4, dep))`. Could you change the test name?

Reply:
I agree with your suggestion and have renamed the method to `testJoinWithTruncatePartitioning`.